src/meta/src/manager/cluster.rs

```diff
@@ -226,28 +225,47 @@ where
 
     pub async fn update_schedulability(
         &self,
-        host_address: HostAddress,
+        worker_ids: &[u32],
         is_unschedulable: bool,
-    ) -> MetaResult<WorkerType> {
+    ) -> MetaResult<Vec<WorkerType>> {
         let mut core = self.core.write().await;
-        let worker = core
+        let workers = core
             .workers
-            .get_mut(&WorkerKey(host_address.clone()))
-            .ok_or_else(|| anyhow!("Worker node does not exist!"))?;
-        let worker_type = worker.worker_type();
-
-        if let Some(property) = &mut worker.worker_node.property {
-            if property.is_unschedulable == is_unschedulable {
-                return Ok(worker_type);
+            .values_mut()
+            .filter(|w| worker_ids.contains(&w.worker_id()))
+            .collect_vec();
+        if workers.is_empty() {
+            return Err(MetaError::invalid_parameter("Worker nodes do not exist!"));
+        }
+        let worker_types = workers.iter().map(|w| w.worker_type()).collect_vec();
+
+        // return early if any worker has none property OR all already have required schedulability
+        let mut all_correct = true;
+        for worker in &workers {
+            match worker.worker_node.property.as_ref() {
+                None => {
+                    return Err(MetaError::invalid_parameter(
+                        "Worker node does not have property",
+                    ));
+                }
+                Some(prop) => {
+                    if prop.is_unschedulable != is_unschedulable {
+                        all_correct = false;
+                        break;
+                    }
+                }
             }
+        }
+        if all_correct {
+            return Ok(worker_types);
+        }
+
+        for worker in workers {
+            let property = &mut worker.worker_node.property.as_mut().unwrap();
             property.is_unschedulable = is_unschedulable;
-        } else {
-            return Err(MetaError::invalid_parameter(
-                "Worker node does not have property",
-            ));
+            Worker::insert(worker, self.env.meta_store()).await?;
```

Comment:
Updating worker status separately may cause some workers to fail to update when failover encountered. We'd better to introduce a transactional updates for it, but it's still acceptable since the interface is retryable. Just FYI.

Reply:
I agree. Already discussed this with @shanicky. I agree that adding transactions will be useful. Opened an issue here https://github.com/risingwavelabs/risingwave/issues/10598